Assets/Scripts/Model/Building.cs

```diff
@@ -98,6 +133,88 @@ public void Repair()
         GameObject.FindGameObjectWithTag("Human").GetComponent<Player>().SpendCoins((int)repairCost);
         health.AddHealth(repairQuantity);
         NotifyHUD();
+        ApplyMainTexture();
+        ApplySmokeEffect();
         Debug.Log("BUILDING REPAIRED, CURRENT HP: " + health.GetCurrentHealth());
     }
+
+    /*
+     * Applies texture to ub depending on its Health and Level
+     * 
+     */
+    public void ApplyMainTexture()
+    {
+        var hp = health.GetCurrentHealthPercentage();
+        var text = textures[0]; // to avoid null values
+        switch (currentLevel)
```

Comment:
You could even use your enum of HPThreshHold to have a method that assigns a percentage to one of the values of the enum